%%
% (1) remove experiments w/less than N_reads_THRESH reads
%   also remove genes that always have zero reads
% (2) remove experiments with w/correlation w/Nag or vanDijk < CORR_THRESH
% (3) Get relative position to the end of the chr for each gene 
% (4) calc fraction of experiments w/no expression & save table

CORR_THRESH = 0.5  ; % remove experiments w/poor correlation to known decent RNA-seq data
N_reads_THRESH = 1e5 ; % remove experiments w/few reads 
DATASAVEDIR = '~/Develop/Mendoza__ReplicationEvolution/Data/';

%%
load ('~/Downloads/DEE_Expression_Data.mat'); % 300Mb, generated by Miki; ~/Google Drive/Careylab/random RNA library/Promoter effect Manuscript/GitHub_repository/Data/
T = dataset2table( DS_raw ) ;
clear 'DS_raw' 'DS_cor' 'DS_cor_zsc' ;
T = T(:,1:6403); % remove all but experiment names & ORF

%% (1)  remove experiments w/ < N_reads_THRESH reads
vn = T.Properties.VariableNames ;
vn_expidx = find(regexpcmp( vn , '^.RR\d')) ;
vn_notexpidx = find(~regexpcmp( vn , '^.RR\d')) ;
N = NaN( numel(vn_expidx),1);
for I = 1:numel(vn_expidx)
    N(I) = sum( T.(vn{vn_expidx(I)}));
end
T = T( : , [vn_notexpidx vn_expidx( N > N_reads_THRESH ) ]);

%% 1b. remove genes that have zero expression in all experiments
exprmat = table2array( T( : , regexpcmp(T.Properties.VariableNames , '^[SE]RR\d'))); 
always_zero = mean(exprmat==0 , 2)  == 1 ; 
T = T( ~always_zero , : ) ;

%% (2) remove experiments w/low corr to vanDijk15 & Nag08
% A build vanDijk table & join
% 


vanDijk15 = readtable('~/Google Drive/CareyLab/ExternalData/vanDijk15/FitFlow__Fast_Slow.tab','FileType','text','Delimiter','\t');
vanDijk15.Properties.VariableNames = ...
    {'ORF' 'GENE' 'pos' 'TPMfast' 'TPMslow'} ;
vanDijk15.TPM = nanmean( [ vanDijk15.TPMfast vanDijk15.TPMslow ] , 2) ;
vanDijk15.TPM = vanDijk15.TPM + min(vanDijk15.TPM(vanDijk15.TPM>0)) ; % add epsilon to get rid of 0s
vanDijk15 = vanDijk15( : , [1 2 end]);

Nag08 = readtable('~/Google Drive/CareyLab/ExternalData/Nagalakshmi08/data.tab','FileType','text','Delimiter','\t');
Nag08 = Nag08( : , {'Name' 'Transcription_level_log2_'});
Nag08.Properties.VariableNames  = {'ORF' 'Nag08_log2'} ; 

T = outerjoin( T , vanDijk15 , 'Key' ,'ORF','MergeKeys',true);
T = outerjoin( T , Nag08 , 'Key' ,'ORF','MergeKeys',true);

% B remove low corr datasets
vn = T.Properties.VariableNames ;
vn_expidx = find(regexpcmp( vn , '^.RR\d')) ;
vn_notexpidx = find(~regexpcmp( vn , '^.RR\d')) ;
c = NaN( numel(vn_expidx),2);
for I = 1:numel(vn_expidx)
    c(I,1) = corr( log2(T.TPM)  , log2( T.(vn{vn_expidx(I)}) + .1) ,'rows','complete');
    c(I,2) = corr( T.Nag08_log2  , log2( T.(vn{vn_expidx(I)}) + .1) ,'rows','complete');
end
T = T( : , [vn_notexpidx vn_expidx( all(c>CORR_THRESH,2)  ) ]);


%% (3) Get relative position to the end of the chr for each gene 
%   calc dist from end of chr
SGD = dataset2table( loadSGDFeaturesDS() );
SGD.Chr = str2double(SGD.Chr);
chrlengths = readtable('~/Google Drive/CareyLab/ExternalData/Yeast/genome.chrlengths','FileType','text','Delimiter','\t');
chrlengths.Chr = NaN( height(chrlengths),1);
chrlengths.Chr(1:17) = 1:17 ;
chrlengths = chrlengths(1:17,:);
chrlengths.Properties.VariableNames = {'Chrom' 'chrlen' 'Chr'};
T = innerjoin( T , SGD , 'Key','ORF');
T = innerjoin( T , chrlengths(:,{'chrlen' 'Chr'}) , 'Key','Chr');
T = sortrows(T,{'Chr' 'Start'});
dist_to_end = T.chrlen - max([T.Start T.End],[],2) ;
dist_to_begin =  min([T.Start T.End],[],2) ;
T.dist_to_end = min( [ dist_to_end dist_to_begin ],[],2) ;

%% (4) calc fraction of experiments w/no expression
% also median expression and median expression when there are >0 read counts
exprmat = table2array( T( : , regexpcmp(T.Properties.VariableNames , '^[SE]RR\d'))); 
T.expr_l2 = log2( median(exprmat,2) + 1.01);
T.expr_l2_nonzero = log2( arrayfun( @(I) median( exprmat(I, exprmat(I,:)>0)) , 1:height(T) )' ) ;
T.fraction_experiments_zero_reads = mean(exprmat==0 , 2) ;

DEE = T ; 
save( [DATASAVEDIR 'DEE.mat' ] , 'DEE' );
